app/controllers/admin/data_update_scripts_controller.rb

```diff
@@ -3,7 +3,20 @@ class DataUpdateScriptsController < Admin::ApplicationController
     layout "admin"
 
     def index
-      @data_update_scripts = DataUpdateScript.all
+      @data_update_scripts = DataUpdateScript.order(run_at: :desc)
+    end
+
+    def show
+      response = DataUpdateScript.find(params[:id])
+      render json: { response: response }
+    rescue ActiveRecord::RecordNotFound => e
+      render json: { error: "#{e.class}: #{e.message}" }, status: :not_found
+    end
+
+    def force_run
+      DataUpdateWorker.perform_async(params[:id])
+    rescue StandardError => e
+      render json: { error: "#{e.class}: #{e.message}" }, status: :unprocessable_entity
```

Comment:
I'm not sure there's a scenario when `StandardError` can be triggered here as you're executing an async script. The error inside the script will be triggered by the script, not by the controller:

```ruby
pry(main)> DataUpdateWorker.perform_async("abcdef")
=> "2fe553e19238c6d3ec7471c9"
```

you can pass whatever, it's Sidekiq that's going to fail when it deserializes an invalid ID

Reply:
@rhymes Do you think that the rescue block in this case in unnecessary then?

Reply:
@Ridhwana yeah. there's no scenario when `.perform_async` fails. Even if you don't pass required articles to it it won't fail. It will fail when the job is picked up by Sidekiq